python/tests/test_nn.py

```diff
@@ -19,6 +19,28 @@ def test_linear(self):
         outputs = layer(inputs)
         self.assertEqual(tuple(outputs.shape), (10, 8))
 
+    def test_int8_quantize(self):
+        inputs = mx.full((2, 2), 2.0, dtype=mx.float32)
+        new_inputs, scale = nn.absmax_quantize(inputs)
+        res = new_inputs * scale
+        assert np.allclose(res, inputs)
+
+    def test_linear_quantize(self):
+        inputs = mx.full((2, 2), 2.0, dtype=mx.float32)
+        layer = nn.AbsmaxQuantizedLinear(input_dims=2, output_dims=2)
+        outputs = layer(inputs)
+        self.assertEqual(tuple(outputs.shape), (2, 2))
+
+        x = mx.full((2, 2), 10, dtype=mx.float32)
+        setattr(layer, "weight", x)
+        self.assertEqual(layer.weight.dtype, mx.int8)
+        self.assertEqual(tuple(layer.weight.shape), (2, 2))
+        res = layer(x)
+        self.assertEqual(res.dtype, mx.float32)
+        self.assertEqual(tuple(res.shape), (2, 2))
+        # note the previous test set the weight to 10 so its 10 @ 10
+        self.assertEqual(res.tolist(), mx.full((2, 2), 200, dtype=mx.float32).tolist())
+
```

Comment:
No tests check for error cases like incorrect input shapes/types or unsupported weight types. Adding negative tests is important.

Reply:
is the desire to add asserts for dtype/shape inside the call function of the module?